src/Entity/Portal.php

```diff
@@ -1118,4 +1118,128 @@ function save() {
         $this->_save($manager);
         $this->_changes = array();
     }
+
+
+    ###################################################
+    # archiving and deleting rooms
+    ###################################################
+
+    function setStatusArchivingUnusedRooms(bool $statusArchivingUnusedRooms) {
+        $this->extras['ARCHIVING_ROOMS_STATUS'] = $statusArchivingUnusedRooms ? 1 : -1;
+        return $this;
+    }
+
+    public function getStatusArchivingUnusedRooms(): bool
+    {
+        $statusArchivingUnusedRooms = $this->extras['ARCHIVING_ROOMS_STATUS'] ?? -1;
+        return $statusArchivingUnusedRooms === 1;
+    }
+
+
+    public function turnOnArchivingUnusedRooms() {
+        $this->extras['ARCHIVING_ROOMS_STATUS'] = 1;
+    }
+
+    public function turnOffArchivingUnusedRooms() {
+        $this->extras['ARCHIVING_ROOMS_STATUS'] = -1;
+    }
+
+    public function getDaysUnusedBeforeArchivingRooms():int
+    {
+        $retour = 365; //default
+        if ($this->_issetExtra('ARCHIVING_ROOMS_DAYS_UNUSED_BEFORE_ARCHIVE')) {
+            $retour = $this->extras['ARCHIVING_ROOMS_DAYS_UNUSED_BEFORE_ARCHIVE'];
+        }
+        return $retour;
+    }
+
+    public function setDaysUnusedBeforeArchivingRooms (int $value) {
+        $this->extras['ARCHIVING_ROOMS_DAYS_UNUSED_BEFORE_ARCHIVE'] = $value;
+    }
+
+    public function isActivatedArchivingUnusedRooms():bool
+    {
+        $status = $this->getStatusArchivingUnusedRooms();
+        return $status === 1;
+    }
+
+    /** get days send an email before archiving an unused room
+     *
+     * @return int days send email before archiving an unused room
+     */
+    public function getDaysSendMailBeforeArchivingRooms ():int
+    {
+        $retour = 0;
+        if ($this->_issetExtra('ARCHIVING_ROOMS_DAYS_SEND_MAIL_BEFORE_ARCHIVE')) {
+            $retour = $this->extras['ARCHIVING_ROOMS_DAYS_SEND_MAIL_BEFORE_ARCHIVE'];
+        }
+        return $retour;
+    }
```

Comment:
```suggestion
    public function getDaysSendMailBeforeArchivingRooms(): int
    {
        $retour = 0;
        if ($this->_issetExtra('ARCHIVING_ROOMS_DAYS_SEND_MAIL_BEFORE_ARCHIVE')) {
            $retour = $this->extras['ARCHIVING_ROOMS_DAYS_SEND_MAIL_BEFORE_ARCHIVE'];
        }
        return $retour;
    }
```

Besser wäre wie bei den anderen Gettern auch auf _issetExtra zu verzichten.